plotly/tests/test_core/test_plotly/test_plot.py

```diff
@@ -66,3 +70,187 @@ def test_plot_empty_data(self):
         py.sign_in('PlotlyImageTest', '786r5mecv0')
         self.assertRaises(PlotlyEmptyDataError, py.plot, [],
                           filename='plot_invalid')
+
+    def test_plot_sharing_argument(self):
+
+        # Raise an error if sharing argument is incorrect
+        # correct arguments {'public, 'private', 'secret'}
+
+        fig = {
+            'data': [
+                {
+                    'x': [1, 2, 3],
+                    'y': [2, 1, 2]
+                }
+            ]
+        }
+
+        kwargs = {'filename': 'invalid-sharing-argument',
+                  'sharing': 'privste'}
+
+        with self.assertRaisesRegexp(PlotlyError, 'sharing'):
+            py.plot(fig, **kwargs)
+
+    def test_plot_world_readable_sharing_conflict_1(self):
+
+        # Raise an error if world_readable=False but sharing='public'
+
+        fig = {
+            'data': [
+                {
+                    'x': [1, 2, 3],
+                    'y': [2, 1, 2]
+                }
+            ]
+        }
```

Comment:
ditto https://github.com/plotly/python-api/pull/264/files#r36833746

8 lines could be on 1 with similar readability, imo.